### ISSUE TITLE
Cannot specify server-supplied enchantments(EcoEnchants) in cits

### ISSUE THREAD

**Issue by RevengeLordOfMaj** [SHsuperCM/CITResewn#27]
I am trying to use this with the EcoEnchants plugin, but items that have an enchant from the plugin applied don't seem to change their texture. Works fine with opti, and I'm not sure what other things break due to this.

**Comment by SHsuperCM** [comment]
Can you send the resourcepack and your logs?

**Comment by RevengeLordOfMaj** [comment]
sorry, where would the logs be? here's the pack:
[MajPack.zip](https://github.com/SHsuperCM/CITResewn/files/7182205/MajPack.zip)

**Comment by SHsuperCM** [comment]
Have the resourcepack enabled and upload the contents of `.minecraft/logs/latest.log` to something like [pastebin](https://pastebin.com/)

**Comment by RevengeLordOfMaj** [comment]
here: https://pastebin.com/ngak9H26

**Comment by SHsuperCM** [comment]
I think you sent me the wrong resourcepack, it doesnt have `minecraft:optifine/cit/zeldastuff/charge.properties`. 
Also, the problem is that the client doesn't have the enchantments registered when the resourcepack is loaded, it'll take a little longer to fix(probably after I close #19)

**Comment by RevengeLordOfMaj** [comment]
ah, might have sent an older version, here:
[MajPack.zip](https://github.com/SHsuperCM/CITResewn/files/7182342/MajPack.zip)
and no rush, thanks for being willing to help :)

**Comment by SirSebi** [comment]
That's the problem I also opend an issue. That's cause ecoenchants change the way of naming the items

**Comment by SirSebi** [comment]
Also I tried to change the prefix to * CUSTOM NAME * but that's no fix so the only fix is removing ecoenchants. But that's not capable for me cause it's a payed plugin :D

**Comment by SHsuperCM** [comment]
@SirSebi 
Thanks! Good to know the cause, but also this is a different issue, I'll add compatibility for the custom enchantments supplied by the server but it means the client will do it blindly because it cant know what plugins are in the server.
I'll probably turn the enchantment load error into a warning and make the item check not [go through the client's enchantments registry](https://github.com/SHsuperCM/CITResewn/blob/dc14c845024e78a0832acad0ab62f9ce781aa259/src/main/java/shcm/shsupercm/fabric/citresewn/pack/cits/CIT.java#L270).

**Comment by SHsuperCM** [comment]
The issue here is this btw:
```
[22:02:31] [Worker-Main-10/ERROR]: Skipped CIT: Unknown enchantment minecraft:charge in MajPack -> minecraft:optifine/cit/zeldastuff/charge.properties
```

**Comment by RevengeLordOfMaj** [comment]
Actually, although that was one of the issues I had, I believe the main issue was something else. Any item that has a enchant from the plugin seems to not work at all, even if the texture is based off of the item's name, enchants supplied by the plugin revert items to their original texture. I apologize if this is under the same fix.

**Comment by SirSebi** [comment]
Yeah look at the issue I linked. It's because ecoenchants changes the data of the item from Name:'{"text":"CUSTOM MAME}' to Name:'{"extra":[{"text":"CUSTOM NAME"}],"text":""}'

because of that, the mod can't apply the cit

**Comment by SHsuperCM** [comment]
No its not @SirSebi

**Comment by SHsuperCM** [comment]
@RevengeLordOfMaj When checking the items it will use the client's registry to see if any enchantments match, I'll need to make it check the item's nbt for enchantments blindly without the client's registry

**Comment by SHsuperCM** [comment]
https://github.com/SHsuperCM/CITResewn/blob/dc14c845024e78a0832acad0ab62f9ce781aa259/src/main/java/shcm/shsupercm/fabric/citresewn/pack/cits/CIT.java#L270

In line 270 it uses something from vanilla `EnchantmentHelper` which works great if both the server and the client have the same registry, the mod just needs to avoid that because sometimes the server might have something else.

**Comment by SHsuperCM** [comment]
> The issue here is this btw:
> 
> ```
> [22:02:31] [Worker-Main-10/ERROR]: Skipped CIT: Unknown enchantment minecraft:charge in MajPack -> minecraft:optifine/cit/zel
> ```

Apologies for this btw, while it is also a problem from similar origins, it is just a bit different from the one I just mentioned.
They'd be fixed together.

**Comment by RevengeLordOfMaj** [comment]
Aight, ty :))

**Comment by SirSebi** [comment]
So in my case its also not the problem that ecoenchants changes the way of naming items?

**Comment by SHsuperCM** [comment]
No, I still believe that's a separate issue. I am not 100% sure that this plugin is the cause but something in your server caused the item you did the `/data` on to have the extra json parts.
Here though it is very clearly about the custom enchantments themselves because the issue only happens with the enchants being present in the item.

**Comment by SHsuperCM** [comment]
Fixed in [v0.6.0](https://modrinth.com/mod/cit-resewn/version/TaegaFdJ)

**Comment by RevengeLordOfMaj** [comment]
uhh about that...
![2021-09-18_11 01 53](https://user-images.githubusercontent.com/90877042/133895110-672528b4-aded-45a9-98e6-a16ee7191f9b.png)

**Comment by SHsuperCM** [comment]
I tested the charge enchantment cit from your pack and it worked on my end, Imma need to see some files if something isnt working.

**Comment by RevengeLordOfMaj** [comment]
looked at the charge one and it seemed to be working. Here's new log: https://pastebin.com/exh9d0bs

**Comment by RevengeLordOfMaj** [comment]
oh and here's the nbt for the helmet just in case: https://pastebin.com/ukRuRqks
Might be too much data for it to handle?

**Comment by SHsuperCM** [comment]
Can I see the `properties` for the cit that isnt loading?

**Comment by RevengeLordOfMaj** [comment]
type=item
items=minecraft:netherite_helmet
nbt.display.Name=ipattern:*Crystalite*

**Comment by SHsuperCM** [comment]
Ah.. Yes, that's a different issue(probably also caused by EcoEnchants) that I'm afraid I cant help with right now with where the mod is at(without #31 being finished).
Have a read through #25 to see why and how you can make that cit work.

**Comment by RevengeLordOfMaj** [comment]
Alr, ty :)

**Comment by RevengeLordOfMaj** [comment]
oh and mb on the .properties one, the wildcards turned the name to italics

**Comment by SHsuperCM** [comment]
No problem, I saw that. Still without the `{` in the pattern, it will try to add the vanilla anvil json wrapper to the pattern. so you'd have to use `{*Crystalite*}` for it to apply for now.

**Comment by RevengeLordOfMaj** [comment]
aight, thanks again :)

**Comment by SHsuperCM** [comment]
Fixed the json name issue in [v0.7.0](https://modrinth.com/mod/cit-resewn/version/S0QSA59b)
